### DISCUSSION TITLE
VS Code Setup for Module.c module.h org

### DISCUSSION THREAD

**Discussion by prasannabe2004** [Mair/esp32-course#115] [Course Qeustions]
Hi I am following the IDF Components - Other Source files. When i move the module.h to components/mymodule directory, module.h reference in main.c show error, but the help does not provide me option to add the mymodule to the include path. i tried manually but i am still stuck with undefined reference to mymodule function definition. Am I missing anything in VS code setup or cmake to link that function?

Option it show are 
Disable error Squiggles
Learn how to install a library for this header with vcpkg

**Comment by Mair** [comment]
try to compile and then restart vs code.
Let me know if the issue persists.

**Reply by prasannabe2004** [reply]
Yes compile/restart/uninstall/install vscode etc... didnt help. Initially i was hesitant to edit cmake files. But I modified the cmakefile manually and get this does for now. Not sure i did correctly or not though.

**Reply by Mair** [reply]
what did you change?

**Reply by prasannabe2004** [reply]
I modified cmake files in main directory and components/mymodule to include the path
